### PR TITLE
[TASK] Update information on content rendering

### PR DESCRIPTION
**Pull request by linawolf**
Move information on jumbotron, subpage layout and adding content to their own pages. Those contents have not been changed yet.

Releases: main, 13.4

### CODE REVIEW

File: Documentation/ContentMapping/Index.rst [hunk 1]
```diff
@@ -36,90 +31,39 @@ Your site set configuration should now look like this:
...
-                               }
-                           }
-                       }
-                   }
-               }
-           }
-       }
-   }
+Each area in the page layout becomes an identifier that can be used during
+content mapping. If not content element is added in the backend of that page and
```
Comment (froemken) [inline Documentation/ContentMapping/Index.rst:62]:
```suggestion
content mapping. If no content element is added in the backend of that page and
```

File: Documentation/ContentMapping/Index.rst [hunk 2]
```diff
@@ -131,323 +75,131 @@ properties at :guilabel:` Appearance >  Page Layout > Backend Layout`.
 
 .. include:: /Images/AutomaticScreenshots/ChooseBackendLayout.rst.txt
```
Comment (froemken) [inline Documentation/ContentMapping/Index.rst:76]:
```suggestion
..  include:: /Images/AutomaticScreenshots/ChooseBackendLayout.rst.txt
```

File: Documentation/ContentMapping/Index.rst [hunk 3]
```diff
@@ -131,323 +75,131 @@ properties at :guilabel:` Appearance >  Page Layout > Backend Layout`.
...
+:ref:`Load the example data automatically <t3sitepackage:load-example-data>`
+the areas "Jumbotron" and "Main" should already contain some example content.
 
-.. highlight:: html
-
-Open the file :file:`Resources/Private/Templates/Page/Default.html` and locate the
-main content area. It contains a headline (look for the :code:`<h2>`-tags) and
-some dummy content (look for the :code:`<p>`-tags).
+In the database each content element record is stored in the table
+:sql:`tt_content`. This table has a filed called `colPos`. If the value stored
```
Comment (froemken) [inline Documentation/ContentMapping/Index.rst:92]: Context is database and table. So I prefer using:

```suggestion
:sql:`tt_content`. This table has a column called `colPos`. If the value stored
```

File: Documentation/ContentMapping/Index.rst [hunk 4]
```diff
@@ -131,323 +75,131 @@ properties at :guilabel:` Appearance >  Page Layout > Backend Layout`.
...
+in field `colPos` is the same as defined in the page layout in page TSconfig
```
Comment (froemken) [inline Documentation/ContentMapping/Index.rst:93]:
```suggestion
in column `colPos` is the same as defined in the page layout in page TSconfig
```

File: Documentation/ContentMapping/Index.rst [hunk 5]
```diff
@@ -131,323 +75,131 @@ properties at :guilabel:` Appearance >  Page Layout > Backend Layout`.
...
-
-.. _cm-Preview-page:
-
-Preview page
-============
-
-We have made changes to the Fluid templates of the extension above. It is
-therefore necessary to :guilabel:`Flush the content caches` in the Menu in the
-top bar before you can preview the page properly:
+.. _cm-fluid-typoscript-mapping:
```
Comment (froemken) [inline Documentation/ContentMapping/Index.rst:148]:
```suggestion
..  _cm-fluid-typoscript-mapping:
```

File: Documentation/ContentMapping/Index.rst [hunk 6]
```diff
@@ -131,323 +75,131 @@ properties at :guilabel:` Appearance >  Page Layout > Backend Layout`.
...
 
-You can switch the used page backend layout in the page properties at
-:guilabel:` Appearance >  Page Layout > Backend Layout`. Edit the page
-properties of your page to use the backend layout "Two Columns".
+Extract the content element rendering to a partial
+==================================================
 
-.. include:: /Images/AutomaticScreenshots/SwitchBackendLayout.rst.txt
+As we want to reuse the Fluid part about rendering content elements in the
+next steps, we extract it into a partial, like we die with the menu in
```
Comment (froemken) [inline Documentation/ContentMapping/Index.rst:180]:
```suggestion
next steps, we extract it into a partial, like we did with the menu in
```

File: Documentation/ContentMapping/Index.rst [hunk 7]
```diff
@@ -131,323 +75,131 @@ properties at :guilabel:` Appearance >  Page Layout > Backend Layout`.
...
 
-Enter some content to the sidebar. You could for example use the content element
-"Menu of subpages" to display a menu in the sidebar.
+The partial then looks like this:
 
-Preview the page once more. A sidebar will appear in the frontend:
+..  include:: /CodeSnippets/Fluid/PartialContent.rst.txt
 
-.. include:: /Images/AutomaticScreenshots/TwoColumnsPreviewPage.rst.txt
+.. _content-element-next-steps:
```
Comment (froemken) [inline Documentation/ContentMapping/Index.rst:194]:
```suggestion
..  _content-element-next-steps:
```

File: Documentation/ContentMapping/Jumbotron.rst [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+
+.. code-block:: html
+   :caption: theme/index.html
+
+   <div class="jumbotron">
+       <div class="container">
+           <h1 class="display-3">Hello, world!</h1>
+           <p> ... </p>
+       </div>
+   </div>
```
Comment (froemken) [inline Documentation/ContentMapping/Jumbotron.rst:21]:
```suggestion
..  code-block:: html
    :caption: theme/index.html

    <div class="jumbotron">
        <div class="container">
            <h1 class="display-3">Hello, world!</h1>
            <p> ... </p>
        </div>
    </div>
```

File: Documentation/ContentMapping/Jumbotron.rst [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+
+We could use a content element of type :guilabel:`Text` to store the needed
+information. However the output of the Standard content element "Text" look
```
Comment (froemken) [inline Documentation/ContentMapping/Jumbotron.rst:24]:
```suggestion
information. However the output of the default content element "Text" looks
```

File: Documentation/ContentMapping/Jumbotron.rst [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+   :caption: Example HTML Output
+
+   <div id="c215" class="frame frame-default frame-type-text frame-layout-0">
+      <header>
+         <h2 class="">
+            Hello World!
+         </h2>
+      </header>
+      <p>Lorem ipsum dolor sit amet, ...</p>
+   </div>
```
Comment (froemken) [inline Documentation/ContentMapping/Jumbotron.rst:37]:
```suggestion
..  code-block:: html
    :caption: Example HTML Output

    <div id="c215" class="frame frame-default frame-type-text frame-layout-0">
        <header>
            <h2 class="">
                Hello World!
            </h2>
        </header>
        <p>Lorem ipsum dolor sit amet, ...</p>
    </div>
```

File: Documentation/ContentMapping/Jumbotron.rst [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+content we will render it ourselves in this partial. Add the debug view helper
+to the partial to see what the data of the jumbotronContent looks like:
+
+.. code-block:: html
+   :caption: EXT:site_package/Resources/Private/Partials/Page/Jumbotron.html
+
+   <div class="jumbotron">
+      <f:debug>{jumbotronContent}</f:debug>
+      <!-- ... -->
+   </div>
```
Comment (froemken) [inline Documentation/ContentMapping/Jumbotron.rst:56]:
```suggestion
..  code-block:: html
    :caption: EXT:site_package/Resources/Private/Partials/Page/Jumbotron.html

    <div class="jumbotron">
        <f:debug>{jumbotronContent}</f:debug>
        <!-- ... -->
    </div>
```

File: Documentation/ContentMapping/Jumbotron.rst [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+   :caption: EXT:site_package/Resources/Private/Partials/Page/Jumbotron.html
+
+   <div class="jumbotron">
+       <f:for each="{jumbotronContent}" as="contentElement">
+           <div class="container">
+               <h1 class="display-3">{contentElement.data.header}</h1>
+               <f:format.html>{contentElement.data.bodytext}</f:format.html>
+           </div>
+       </f:for>
+   </div>
```
Comment (froemken) [inline Documentation/ContentMapping/Jumbotron.rst:72]:
```suggestion
..  code-block:: html
    :caption: EXT:site_package/Resources/Private/Partials/Page/Jumbotron.html

    <div class="jumbotron">
        <f:for each="{jumbotronContent}" as="contentElement">
            <div class="container">
                <h1 class="display-3">{contentElement.data.header}</h1>
                <f:format.html>{contentElement.data.bodytext}</f:format.html>
            </div>
        </f:for>
    </div>
```

File: Documentation/ContentMapping/SubpageLayout.rst [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+.. _cm-switch_backend_layout:
+
+Switch to the two column layout with a sidebar
+==============================================
+
+You can switch the used page backend layout in the page properties at
+:guilabel:` Appearance >  Page Layout > Backend Layout`. Edit the page
+properties of your page to use the backend layout "Two Columns".
+
+.. include:: /Images/AutomaticScreenshots/SwitchBackendLayout.rst.txt
```
Comment (froemken) [inline Documentation/ContentMapping/SubpageLayout.rst:28]:
```suggestion
..  include:: /Images/AutomaticScreenshots/SwitchBackendLayout.rst.txt
```

File: Documentation/ContentMapping/SubpageLayout.rst [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+
+After saving you will see that the content of the columns "main" and
+"jumbotron" remains unchanged while there is a third column "sidebar".
+This is due to the fact that the backend layout "Default" and "TwoColumns"
+use the same colPos number for these columns.
+
+.. include:: /Images/AutomaticScreenshots/BackendLayoutTwoColumns.rst.txt
```
Comment (froemken) [inline Documentation/ContentMapping/SubpageLayout.rst:35]:
```suggestion
..  include:: /Images/AutomaticScreenshots/BackendLayoutTwoColumns.rst.txt
```

File: Documentation/ContentMapping/SubpageLayout.rst [hunk 15]
```diff
@@ -0,0 +1,42 @@
...
+
+Enter some content to the sidebar. You could for example use the content element
+"Menu of subpages" to display a menu in the sidebar.
+
+Preview the page once more. A sidebar will appear in the frontend:
+
+.. include:: /Images/AutomaticScreenshots/TwoColumnsPreviewPage.rst.txt
```
Comment (froemken) [inline Documentation/ContentMapping/SubpageLayout.rst:42]:
```suggestion
..  include:: /Images/AutomaticScreenshots/TwoColumnsPreviewPage.rst.txt
```

File: Documentation/ContentMapping/_codesnippets/_SectionMainRenderPartial.diff [hunk 16]
```diff
@@ -0,0 +1,8 @@
+ <f:layout name="Layout"/>
+ <f:section name="Main">
+     <f:render partial="Jumbotron" arguments="{_all}"/>
+     <div class="container">
+-        <f:for each="{content.main.records}" as="record">...</f:for>
++        <f:render partial="Content"  arguments="{records: content.main.records}"/>
```
Comment (froemken) [inline Documentation/ContentMapping/_codesnippets/_SectionMainRenderPartial.diff:6]:
```suggestion
+        <f:render partial="Content" arguments="{records: content.main.records}"/>
```

File: Documentation/ContentMapping/AddContent.rst [hunk 17]
```diff
@@ -0,0 +1,43 @@
...
+
+.. include:: /Images/AutomaticScreenshots/FillNewContentElement.rst.txt
+
+The new content element appears in the appropriate column. Repeat this process
+and enter some further content in the column "Jumbotron". The page module should
+now look like this:
+
+.. include:: /Images/AutomaticScreenshots/ContentMappingPreviewPage.rst.txt
+
+.. _cm-Preview-page:
```
[the remark this answers is not in the document]
Comment (linawolf) [reply Documentation/ContentMapping/AddContent.rst]: It is not nice but not harmful either. We automatically lowercase and sluggify all anchors before they are used.

File: Documentation/ContentMapping/AddContent.rst [hunk 18]
```diff
@@ -0,0 +1,43 @@
...
+
+..  include:: /Images/AutomaticScreenshots/FillNewContentElement.rst.txt
+
+The new content element appears in the appropriate column. Repeat this process
+and enter some further content in the column "Jumbotron". The page module should
+now look like this:
+
+..  include:: /Images/AutomaticScreenshots/ContentMappingPreviewPage.rst.txt
+
+..  _cm-Preview-page:
```
Comment (linawolf) [inline Documentation/ContentMapping/AddContent.rst:30]:
```suggestion
..  _cm-preview-page:
```

File: Documentation/ContentMapping/AddContent.rst [hunk 19]
```diff
@@ -0,0 +1,43 @@
+:navigation-title: Jumbotron
+..  include:: /Includes.rst.txt
+
+.. _cm-typo3-backend-add-content:
```
Comment (froemken) [inline Documentation/ContentMapping/AddContent.rst:4]:
```suggestion
..  _cm-typo3-backend-add-content:
```

File: Documentation/ContentMapping/AddContent.rst [hunk 20]
```diff
@@ -0,0 +1,43 @@
+:navigation-title: Jumbotron
```
Comment (linawolf) [inline Documentation/ContentMapping/AddContent.rst:1]:
```suggestion
:navigation-title: Adding content
```

File: Documentation/ContentMapping/SubpageLayout.rst [hunk 21]
```diff
@@ -0,0 +1,42 @@
+:navigation-title: Jumbotron
```
Comment (linawolf) [inline Documentation/ContentMapping/SubpageLayout.rst:1]:
```suggestion
:navigation-title: Subpage layout
```

**Review by froemken — APPROVED** [review]

**Pull request merged**